src/main/java/org/truffleruby/core/fiber/FiberNodes.java

```diff
@@ -98,38 +102,80 @@ public DynamicObject initialize(DynamicObject fiber, DynamicObject block) {
 
     }
 
+    @CoreMethod(names = "transfer", rest = true)
+    public abstract static class TransferNode extends CoreMethodArrayArgumentsNode {
+
+        @Child private FiberTransferNode fiberTransferNode = FiberTransferNodeFactory.create(null);
+
+        @Specialization
+        public Object resume(VirtualFrame frame, DynamicObject fiber, Object[] args,
+                @Cached("create()") GetCurrentRubyThreadNode getCurrentRubyThreadNode,
+                @Cached("createBinaryProfile()") ConditionProfile sameFiberProfile) {
+
+            Layouts.FIBER.setTransferred(fiber, true);
+
+            final DynamicObject currentThread = getCurrentRubyThreadNode.executeGetRubyThread(frame);
+            final FiberManager fiberManager = Layouts.THREAD.getFiberManager(currentThread);
+            final DynamicObject currentFiber = fiberManager.getCurrentFiber();
+
+            if (sameFiberProfile.profile(currentFiber == fiber)) {
+                // A Fiber can transfer to itself
+                return fiberTransferNode.singleValue(frame, args);
+            }
+
+            return fiberTransferNode.executeTransferControlTo(frame, currentThread, currentFiber, fiber, FiberOperation.TRANSFER, args);
```

Comment:
Would there be any advantage to having this in an else branch to the if statement above (style or performance)?

Reply:
It's the same to Graal, so only a matter of style.
Since the same Fiber case seems the exception I prefer it this way.